evap/staff/views.py

```diff
@@ -144,159 +149,192 @@ def __init__(self):
     return render(request, "staff_semester_view.html", template_data)
 
 
+class EvaluationOperation:
+    email_template_name = None
+    confirmation_message = None
+
+    @staticmethod
+    def applicable_to(evaluation):
+        raise NotImplementedError
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        raise NotImplementedError
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        raise NotImplementedError
+
+
+class RevertToNewOperation(EvaluationOperation):
+    confirmation_message = ugettext_lazy("Do you want to revert the following evaluations to preparation?")
+
+    @staticmethod
+    def applicable_to(evaluation):
+        return evaluation.state in ['prepared', 'editor_approved', 'approved']
+
+    @staticmethod
+    def warning_for_inapplicables(amount):
+        return ungettext("{} evaluation can not be reverted, because it already started. It was removed from the selection.",
+            "{} evaluations can not be reverted, because they already started. They were removed from the selection.", amount).format(amount)
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        for evaluation in evaluations:
+            evaluation.revert_to_new()
+            evaluation.save()
+        messages.success(request, ungettext("Successfully reverted {} evaluation to in preparation.",
+            "Successfully reverted {} evaluations to in preparation.", len(evaluations)).format(len(evaluations)))
+
+
+class RevertToPreparedOperation(EvaluationOperation):
```

Comment:
you could rename all these to `MoveToPreparedOperation` and similar, as the action is not always actually "reverting" the state (e.g. a change from "new" to "prepared")

Reply:
I prefer to change only this one to `Move`. The rest was fine to @janno42 and me.